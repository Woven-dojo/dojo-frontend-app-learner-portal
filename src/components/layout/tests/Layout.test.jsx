import React from 'react';
import renderer from 'react-test-renderer';
import { AppContext } from '@edx/frontend-platform/react';
import { MemoryRouter } from 'react-router-dom';

import { Layout } from '../index';

jest.mock('@edx/frontend-component-footer', () => () => <footer>SiteFooter</footer>);

jest.mock('../../../colors.scss', () => ({
  white: 'white',
  dark: 'gray',
}));

const APP_CONTEXT = {
  authenticatedUser: {
    userId: 1,
    roles: [],
  },
  config: {},
  enterpriseConfig: {
    branding: {
      colors: {
        primary: 'blue',
        secondary: 'purple',
        tertiary: 'gray',
      },
    },
  },
};

describe('Layout', () => {
  it('should render the expected HTML', async () => {
    let tree = null;

    await renderer.act(async () => {
      tree = await renderer.create(
        <MemoryRouter>
          <div>
            <AppContext.Provider value={APP_CONTEXT}>
              <Layout>
                <div>Content</div>
              </Layout>
            </AppContext.Provider>
          </div>
        </MemoryRouter>,
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
