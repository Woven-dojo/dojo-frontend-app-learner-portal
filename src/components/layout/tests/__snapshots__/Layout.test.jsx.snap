// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the expected HTML 1`] = `
<div>
  <header
    className="header"
  >
    <div
      className="px-0 container-mw-lg container-fluid"
    >
      <div
        className="master-header"
      >
        <div
          className="container"
        >
          <div
            className="align-items-center row"
          >
            <div
              className="master-header__desktop col"
            >
              <div
                className="align-items-center row"
              >
                <div
                  className="flex-grow-0 col"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <img
                      alt=""
                      src="test-file-stub"
                    />
                  </a>
                </div>
                <div
                  className="col"
                >
                  
                </div>
              </div>
            </div>
            <div
              className="flex-grow-0 master-header__desktop col"
            >
              <div
                className="master-header__user-menu dropdown"
                data-testid="dropdown"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="btn-avatar pgn__avatar-button-avatar pgn__avatar-button-avatar-sm dropdown-toggle btn btn-tertiary btn-sm"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <img
                    alt=""
                    className="pgn__avatar pgn__avatar-xs"
                    src="test-file-stub"
                  />
                  Menu
                </button>
              </div>
            </div>
            <div
              className="master-header__mobile justify-content-between col"
            >
              <div
                className="flex-grow-0 pl-0 col"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                </a>
              </div>
              <div
                className="master-header__user-menu dropdown"
                data-testid="dropdown"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="btn-icon btn-icon-primary btn-icon-sm pgn__dropdown-toggle-iconbutton"
                  onClick={[Function]}
                  showLabel={true}
                  type="button"
                >
                  <span
                    className="btn-icon__icon-container"
                  >
                    <span
                      className="pgn__icon btn-icon__icon"
                    >
                      <svg
                        aria-hidden={true}
                        fill="none"
                        focusable={false}
                        height={24}
                        role="img"
                        viewBox="0 0 24 24"
                        width={24}
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <main
    id="content"
  >
    <div>
      Content
    </div>
  </main>
  <footer
    className="master-footer dojo-footer"
  >
    <div
      className="container"
    >
      <div
        className="align-items-center row"
      >
        <div
          className="d-flex justify-content-sm-center justify-content-lg-start col-lg-6"
        >
          Copyright 2022 Dojo. All rights reserved
        </div>
      </div>
    </div>
  </footer>
</div>
`;
